pgmpy/factors/Factor.py

```diff
@@ -770,9 +770,14 @@ def __eq__(self, other):
                 return True
 
     def __hash__(self):
-        return hash(str(self.variables) + str(self.cardinality.tolist()) +
-                    str(self.values.astype('float').tolist()))
-
+        value_index = 0
```

Comment:
@khalibartan Check how we change the order of variables in other methods by swapping axes. And before computing the hash simply just sort the variables and then compute the hash.